extensions/wikia/ThemeDesigner/ThemeSettings.class.php

```diff
@@ -117,120 +139,172 @@ public function getHistory() {
 				$entry['settings'] = array_merge( $this->defaultSettings, $entry['settings'] );
 			}
 		} else {
-			$history = array();
+			$history = [];
 		}
 
 		foreach ( $history as $key => $val ) {
-			$history[$key]['settings']['background-image'] = $this->getFreshURL( $val['settings']['background-image-name'], ThemeSettings::BackgroundImageName );
+			$history[$key]['settings']['background-image'] = $this->getFreshURL(
+				$val['settings']['background-image-name'],
+				ThemeSettings::BackgroundImageName
+			);
+			$history[$key]['settings']['community-header-background-image'] = $this->getFreshURL(
+				$val['settings']['community-header-background-image'],
+				ThemeSettings::CommunityHeaderBackgroundImageName
+			);
 		}
 
 		return $history;
 	}
 
-	public function saveSettings( $settings, $cityId = null ) {
-		global $wgCityId, $wgUser;
-		$cityId = empty( $cityId ) ? $wgCityId : $cityId;
+	private function saveImage(
+		array &$settings,
+		string $name,
+		string $title,
+		array $previewThumbnailDimensions = [],
+		bool $setDimensions = false,
+		callable $callback = null
+	) {
+		if ( isset( $settings["{$name}-name"] ) && strpos( $settings["{$name}-name"], 'Temp_file_' ) === 0 ) {
+			$temp_file = new LocalFile( Title::newFromText( $settings["{$name}-name"], NS_FILE ), RepoGroup::singleton()->getLocalRepo() );
+			$file = new LocalFile( Title::newFromText( $title, NS_FILE ), RepoGroup::singleton()->getLocalRepo() );
+			$file->upload( $temp_file->getPath(), '', '' );
+			$temp_file->delete( '' );
 
-		// Verify wordmark length ( CONN-116 )
-		if ( !empty( $settings[ 'wordmark-text' ] ) ) {
-			$settings[ 'wordmark-text' ] = trim( $settings[ 'wordmark-text' ] );
-		}
+			// FIXME: XW-3596 - this is hack
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this "FIXME" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)